### PR TITLE
Use .form-select

### PR DESCRIPTION
**Pull request by ciar4n**
Converts to `form-select` in core.
Allows the use of pre-bs5 classes in 3PD extensions (custom-select etc).

### CODE REVIEW

File: templates/cassiopeia/scss/vendor/bootstrap/_custom-forms.scss [hunk 1]
```diff
@@ -66,7 +66,7 @@
 }
 
 // Backwards compat with BS4
-.custom-select {
+.form-select {
```
Comment (wilsonge) [inline templates/cassiopeia/scss/vendor/bootstrap/_custom-forms.scss]: Pretty sure this part isn't needed :)

File: administrator/templates/atum/scss/blocks/_form.scss [hunk 2]
```diff
@@ -177,7 +177,7 @@ legend {
 }
 
 // Backwards compat with BS4
-.custom-select {
+.form-select {
```
Comment (wilsonge) [inline administrator/templates/atum/scss/blocks/_form.scss]: Again assuming this isn't required anymore

**Comment by wilsonge** [comment]
LGTM. I did something similar without the XML conversion (which of course makes total sense). I assume you're missing the casseopia changes - but if you merge in my branch should be fine.

File: templates/cassiopeia/scss/vendor/bootstrap/_custom-forms.scss (hunk 1 above)
Comment (ciar4n) [reply templates/cassiopeia/scss/vendor/bootstrap/_custom-forms.scss]: My bad :)

Comment (ciar4n) [reply templates/cassiopeia/scss/vendor/bootstrap/_custom-forms.scss]: Maybe I miss-understand... do you want to support the `custom-select` class?

Comment (ciar4n) [reply templates/cassiopeia/scss/vendor/bootstrap/_custom-forms.scss]: Im gonna assume not :)

**Comment by ciar4n** [comment]
@wilsonge should be good now!

File: libraries/src/HTML/Helpers/Select.php [hunk 3]
```diff
@@ -137,8 +137,8 @@ public static function genericlist($data, $name, $attribs = null, $optKey = 'val
 		$id = $options['id'] !== false ? $options['id'] : $name;
 		$id = str_replace(array('[', ']', ' '), '', $id);
 
-		// If if the selectbox contains "custom-select-color-state" then load the JS file
-		if (strpos($attribs, 'custom-select-color-state') !== false)
+		// If if the selectbox contains "form-select-color-state" then load the JS file
```
Comment (Quy) [inline libraries/src/HTML/Helpers/Select.php]:
```suggestion
		// If the selectbox contains "form-select-color-state" then load the JS file
```

File: installation/template/scss/template.scss [hunk 4]
```diff
@@ -33,7 +33,7 @@ body {
   background: var(--atum-bg-light);
 }
 
-.custom-select {
+.form-select {
```
Comment (wilsonge) [inline installation/template/scss/template.scss:36]: I guess revert this one?

File: templates/cassiopeia/scss/vendor/bootstrap/_custom-forms.scss [hunk 5]
```diff
@@ -35,27 +35,23 @@
     }
   }
 
-  // All class names are duplicated custom select and form select for b/c with bootstrap 4
-  &.custom-select-color-state, &.form-select-color-state {
+  &.form-select-success {
```
Comment (wilsonge) [inline templates/cassiopeia/scss/vendor/bootstrap/_custom-forms.scss]: I thought we were leaving in the b/c mapping layer for 3rd parties? Looks like you removed them in the conflicts (same for the class below and casseopia)

Comment (ciar4n) [reply templates/cassiopeia/scss/vendor/bootstrap/_custom-forms.scss]: Apologies, I'm getting a bit confused here with some of the comments. These classes are of little use without https://github.com/wilsonge/joomla-cms/pull/55#discussion_r559739569

I've re-added them with https://github.com/wilsonge/joomla-cms/pull/55/commits/57f096b74a9eac347843a19789d974f4f4aadf23

Also moved the extended `.custom-select` as it needs to be loaded before the custom CSS. 

Tested both `form-select` and `custom-select` and both function correctly.

**Comment by wilsonge** [comment]
Thankyou very much!

**Pull request merged**